### PR TITLE
 [18.01] Support re-execution of immediate PJAs when re-queing jobs 

### LINKED ISSUE #4172: Dataset rename PJA not repeated when retrying failed job
When a workflow step fails and I manually resubmit it (and usually click resume dependencies), the rename PJA is not rerun. It triggers me a bit so I have to delete all the jobs and rerun the workflow. :disappointed_relieved: Is this easy to fix (something I could try) or are the PJAs not accessible at the time the job is retried?

### PR DESCRIPTION
**Pull request by dannon**
Fixes #4172

### CODE REVIEW

File: lib/galaxy/tools/actions/__init__.py [hunk 1]
```diff
@@ -782,7 +793,7 @@ def create_collection(self, output, name, **element_kwds):
         if "elements" in element_kwds:
             elements = element_kwds["elements"]
             if hasattr(elements, "items"):  # else it is ELEMENTS_UNINITIALIZED object.
-                for key, value in elements.items():
+                for _key, value in elements.items():
```
Comment (mvdbeek) [inline lib/galaxy/tools/actions/__init__.py]: Maybe a little nicer to just `for value in elements.values()` ?

File: lib/galaxy/tools/actions/__init__.py [hunk 2]
```diff
@@ -598,7 +609,7 @@ def _wrapped_params(self, trans, tool, incoming, input_datasets=None):
 
     def _get_on_text(self, inp_data):
         input_names = []
-        for name, data in reversed(inp_data.items()):
+        for _name, data in reversed(inp_data.items()):
```
Comment (mvdbeek) [inline lib/galaxy/tools/actions/__init__.py]: Same here, `for data in reversed(data.values()):`

**Review by mvdbeek — APPROVED** [review]
Looks good, thanks for the fix!
On a side-note I think there's an additional way (e.g. https://github.com/galaxyproject/galaxy/blob/release_18.01/test/api/test_workflows.py#L2055) to specify output renaming without PJAs, which most tests during the workflow testing use, those also don't work when resuming jobs, I'll look into that.

File: lib/galaxy/tools/actions/__init__.py (hunk 1 above)
Comment (dannon) [reply lib/galaxy/tools/actions/__init__.py]: Sure.

**Comment by dannon** [comment]
@mvdbeek Let me know if you find anything, but I thought that stuff used PJAs as well -- it'd surprise me to have a completely separate path implementing the naming work.

**Comment by mvdbeek** [comment]
> Let me know if you find anything, but I thought that stuff used PJAs as well -- it'd surprise me to have a completely separate path implementing the naming work.

It does, I was just not setting the test up correctly. I'll have an API test for this in a bit, I'll PR that to your branch.

**Comment by mvdbeek** [comment]
I've added an API test for this functionality at https://github.com/dannon/galaxy/pull/26

**Comment by dannon** [comment]
@mvdbeek Thanks!

**Pull request merged**